IOMC/ParticleGuns/src/CloseByParticleGunProducer.cc

```diff
@@ -127,6 +179,25 @@ void CloseByParticleGunProducer::produce(Event& e, const EventSetup& es) {
       p.setY(momentum.y());
       p.setZ(momentum.z());
     }
+
+    // compute correct path considering magnetic field
+    double pathLength = 0.;
+    const double speed = p.pz() / p.e() * c_light / cm;
+    if (PData->charge()) {
+      const double radius = sqrt(p.px() * p.px() + p.py() * p.py()) * 87.78f;  // cm
```

Comment:
To be extremely picky:
* The math is wrong for loopers, right? Maybe they are not _relevant_ in general, but it would be wise to say something or put some protection or do the math right for all cases.
* Would it make sense to ask the value of the magnetic field and act accordingly? With the code as is, all generations with anything but 3.8T will be wrong.

Reply:
I've added a check for loopers and now the magnetic field is taken from the Event Setup